### ISSUE TITLE
Add consistent handedness to perspective projection

### ISSUE THREAD

**Issue by WeiPhil** [dimforge/nalgebra#806]
Hi,
First of all, I would like to thank you for the amazing work done on the library!
I recently tried out some of the transformations and projections in a project. I wanted it to work in a right-handed coordinate system and I was pleased to see the look_at function implemented both for both handedness. However, the perspective matrix is only left-handed if I'm not mistaken. Could we add something similar to the look_at with a _lh/_rh suffix to both make it clear and allow the user to choose between the two?
I'd be happy to add a pull request if you like the idea. 
Best,
Philippe 

Example modification (at the arrows) by adding a boolean to the new construct (_src/geometry.perspective.rs_):
```    
    /// Creates a new perspective matrix from the aspect ratio, y field of view, and near/far planes.
 -> pub fn new(aspect: N, fovy: N, znear: N, zfar: N, is_lh: bool) -> Self {
        assert!(
            !relative_eq!(zfar - znear, N::zero()),
            "The near-plane and far-plane must not be superimposed."
        );
        assert!(
            !relative_eq!(aspect, N::zero()),
            "The aspect ratio must not be zero."
        );

        let matrix = Matrix4::identity();
        let mut res = Self::from_matrix_unchecked(matrix);

        res.set_fovy(fovy);
        res.set_aspect(aspect);
        res.set_znear_and_zfar(znear, zfar);

        res.matrix[(3, 3)] = N::zero();
  ->    res.matrix[(3, 2)] = if is_lh {  N::one() } else { -N::one() };

        res
    }
```

And adding the following two calls in _src/base/cg.rs_:

```
    /// Creates a new homogeneous matrix for a right-handed perspective projection.
    #[inline]
    pub fn new_perspective_rh(aspect: N, fovy: N, znear: N, zfar: N) -> Self {
        Perspective3::new(aspect, fovy, znear, zfar, false).into_inner()
    }

    /// Creates a new homogeneous matrix for a left-handed perspective projection.
    #[inline]
    pub fn new_perspective_lh(aspect: N, fovy: N, znear: N, zfar: N) -> Self {
        Perspective3::new(aspect, fovy, znear, zfar, true).into_inner()
    }
```

**Comment by filnet** [comment]
I have opened a similar issue : https://github.com/dimforge/nalgebra/issues/769
And have done some preliminary work but I am blocked by limitations in Rust.

**Comment by WeiPhil** [comment]
I see you mentioned default parameters beeing a blocker, do you think this could somehow be avoided by a builder pattern (which is usually the way to bypass default params) ? 
I will clause this issue as your apparently already on top of it.

**Comment by filnet** [comment]
My PR is stalled and will probably stay like that for a while.
Feel free to reopen this issue.Whatever comes out will be helpful no matter what.

    Le vendredi 4 décembre 2020 à 13:56:40 UTC+1, Philippe Weier <notifications@github.com> a écrit :  
 
 


Closed #806.
